src/data/user.rs

```diff
@@ -28,6 +28,19 @@ pub struct UserPublicMetrics {
     pub listed_count: usize,
 }
 
+fn empty_string_is_none<'de, D>(deserializer: D) -> Result<Option<Url>, D::Error>
```

Comment:
maybe you can stick this into utils and add a generic T to make it more reusable?

```rust
fn empty_string_is_none<'de, D, T>(deserializer: D) -> Result<Option<T>, D::Error>
  where:
    D: serde::Deserializer<'de>,
    T: std::str::FromStr,
    T::Err: std::fmt::Display
```